library/src/main/java/org/xmtp/android/library/frames/FramesTypes.kt

```diff
@@ -0,0 +1,103 @@
+package org.xmtp.android.library.frames
+
+typealias AcceptedFrameClients = Map<String, String>
+
+sealed class OpenFrameButton {
+    abstract val target: String?
+    abstract val label: String
+
+    data class Link(override val target: String, override val label: String) : OpenFrameButton()
+
+    data class Mint(override val target: String, override val label: String) : OpenFrameButton()
+
+    data class Post(override val target: String?, override val label: String) : OpenFrameButton()
+
+    data class PostRedirect(override val target: String?, override val label: String) : OpenFrameButton()
+}
+
+data class OpenFrameImage(
+    val content: String,
+    val aspectRatio: AspectRatio?,
+    val alt: String?
+)
+
+enum class AspectRatio(val ratio: String) {
+    RATIO_1_91_1("1.91.1"),
+    RATIO_1_1("1:1")
+}
+
+data class TextInput(val content: String)
+
+data class OpenFrameResult(
+    val acceptedClients: AcceptedFrameClients,
+    val image: OpenFrameImage,
+    val postUrl: String?,
+    val textInput: TextInput?,
+    val buttons: Map<String, OpenFrameButton>?,
+    val ogImage: String,
+    val state: String?
+)
+
+data class GetMetadataResponse(
```

Comment:
It's going to make usability much better if this also included the `frameInfo` field which does a lot of the parsing of the extracted tags for you and assembles it into a nice object. Could happen in a follow-up PR.